BPCompatibleAlertControllerDemo/CompatibleAlertControllerDemo/ViewController.swift

```diff
@@ -9,52 +9,69 @@
 import UIKit
 
 class ViewController: UIViewController {
-    var alertController: BPCompatibleAlertController?
 
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        var observerObject: NSObjectProtocol?
+        
+        let cleanupAfterAlertDismissed: (Void) -> Void =
```

Comment:
Yes, I think it's more that NSNotificationCenter is requiring that. My understanding of `NSNotificationCenter`'s `addObserverForName:...` method is that you must call `removeObserver` before you destroy the observer. (Have a look at the Discussion section of [Apple's documentation](https://developer.apple.com/library/ios/documentation/Cocoa/Reference/Foundation/Classes/NSNotificationCenter_Class/#//apple_ref/occ/instm/NSNotificationCenter/addObserverForName:object:queue:usingBlock:)) Happy to be corrected if I've misunderstood something!

As the alert view controller (as used in the demo code) will be deallocated when any action completes, a common clean up function  `cleanupAfterAlertDismissed ` to remove the observer seemed like a way to reduce code duplication. If the demo code didn't use `addObserverForName` there would be no need for this.

Open to suggestions if you can think of a better way around this...